### PR TITLE
Adding descriptions for internal methods of LSP7 & LSP8 

### PR DESCRIPTION
**Pull request by b00ste**
## What does this PR introduce?
In this PR we add description of the relevant internal methods of LSP7 & LSP8 in the “Smart Contract ABI” section of the docs.

## Motivation
Those descriptions can be used by people to extend the internal methods inside some custom methods of their choice.

### CODE REVIEW

File: docs/standards/smart-contracts/lsp7-digital-asset.md [hunk 1]
```diff
@@ -265,6 +265,206 @@ _Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully tra
...
+
+```
+function _burn(
+    address from,
+    uint256 amount,
+    bytes memory data
+) internal virtual
+```
+
+Destroys `amount` tokens.
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp7-digital-asset.md:354]: Lets use other words than destroy

File: docs/standards/smart-contracts/lsp7-digital-asset.md [hunk 2]
```diff
@@ -265,6 +265,206 @@ _Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully tra
 
 :::
 
+## Internal Functions
+
+:::info Warning
+By deploying the smart contracts directly the methods _burn and _mint will lack implementantion.
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp7-digital-asset.md:271]: `will lack implementation` is little bit unclear to me, lets use other words.
Maybe:
>  by deploying an LSP7DigitalAsset contract, there will be no public mint or burn function. In order .. extend the smart contract ...

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 3]
```diff
@@ -335,6 +335,204 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
 
 :::
 
+## Internal Functions
+
+:::info Warning
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:340]: Same for the one above

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 4]
```diff
@@ -335,6 +335,204 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
...
+| `to`     | address  | The receiving address. |
+| `tokenId` | bytes32  | The token to transfer. |
+| `force`  | bool     | When set to TRUE, `to` may be any address; when set to FALSE `to` must be a contract that supports LSP1 UniversalReceiver and not revert. |
+| `memory` | bytes    | Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses. |
+
+:::note
+
+#### Requirements:
+
+- The receiver may revert when the token being sent is not wanted.
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:532]: This is not a requ, lets remove it from here and add it to description

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 5]
```diff
@@ -335,6 +335,204 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
...
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `tokenId` | bytes32  | The token to transfer. |
+| `data`   | bytes    | Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses. |
+
+:::note
+
+#### Requirements:
+
+-
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:500]: empty Requirements, remove please

File: docs/standards/smart-contracts/lsp7-digital-asset.md [hunk 6]
```diff
@@ -265,6 +265,206 @@ _Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully tra
...
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `amount` | uint256  | The amount of token to transfer. |
+| `data`   | bytes    | Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses. |
+
+:::note
+
+#### Requirements:
+
+-
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp7-digital-asset.md:432]: empty requirements

**Review by YamenMerhi — CHANGES_REQUESTED** [review]
Amazing! Let's use ```solidity for code snippets.
Left some comments, and please include `_updateOperator` in lsp7 and `_revokeOperator ` and `_clearOperators` in lsp8 :)

File: docs/standards/smart-contracts/lsp7-digital-asset.md [hunk 7]
```diff
@@ -265,6 +265,221 @@ _Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully tra
...
+```solidity
+function _notifyTokenSender(
+    address from,
+    address to,
+    uint256 amount,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token sender about the `amount` tokens changing owners using LSP1 interface.
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp7-digital-asset.md:417]: An attempt is made to notify the token sender about the `amount` tokens being sent by calling the universalReceiver function on the sender address if it exist.
I guess this sounds better

File: docs/standards/smart-contracts/lsp7-digital-asset.md [hunk 8]
```diff
@@ -265,6 +265,221 @@ _Triggers the **[Transfer](#trasnfer-2)** event when tokens get successfully tra
...
+function _notifyTokenReceiver(
+    address from,
+    address to,
+    uint256 amount,
+    bool force,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token receiver about the `amount` tokens changing owners using LSP1 interface. When force is FALSE the token receiver MUST support LSP1.
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp7-digital-asset.md:440]: An attempt is made to notify the token receiver about the `amount` tokens being received by calling the `universalReceiver(..)` (and link the unviersalReceiver function of the UP) function on the receiver address if it exists.
Sounds better.
Let's also remove this line: `The receiver may revert when the token being sent is not wanted.` because the sender can send and also have a logic where he don't allow any token to be sent.

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 9]
```diff
@@ -335,6 +335,224 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
...
+```solidity
+function _notifyTokenSender(
+    address from,
+    address to,
+    bytes32 tokenId,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token sender about the `tokenId` changing owners using LSP1 interface.
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:485]: same above

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 10]
```diff
@@ -335,6 +335,224 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
...
+
+```solidity
+function _revokeOperator(
+    address operator,
+    address tokenOwner,
+    bytes32 tokenId
+) internal virtual
+```
+
+An attempt is made to notify the token receiver about the `tokenId` changing owners using LSP1 interface. When force is FALSE the token receiver MUST support LSP1.
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:530]: not the right description

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 11]
```diff
@@ -335,6 +335,224 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
...
+
+### _clearOperators
+
+```
+function clearOperators(
+    address tokenOwner,
+    bytes32 tokenId
+) internal virtual
+```
+
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:548]: no description added

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 12]
```diff
@@ -335,6 +335,224 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
...
+|     Name     |   Type   | Description |
+| :----------- | :------- | :---------- |
+| `operator`   | address | The address to revoke as an operator.     |
+| `tokenOwner` | address | The address that is the owner of tokenId. |
+| `tokenId`    | bytes32 | The token to disable operator status to.  |
+
+### _clearOperators
+
+```
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:542]: please use ```solidity

**Review by YamenMerhi — CHANGES_REQUESTED** [review]

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md (hunk 10 above)
Comment (b00ste) [reply docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md]: No description in the natspec, I added custom description.

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md (hunk 12 above)
Comment (b00ste) [reply docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md]: No description in the natspec, I added custom description.

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 13]
```diff
@@ -335,6 +335,226 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
...
+) internal virtual
+```
+
+Revoke the `operator` of the `tokenId` token which belongs to `tokenOwner`.
+
+#### Parameters:
+
+|     Name     |   Type   | Description |
+| :----------- | :------- | :---------- |
+| `operator`   | address | The address to revoke as an operator.     |
```
Comment (CJ42) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:536]:
```suggestion
| `operator`   | address | The address of the operator to revoke.     |
```

**Review by CJ42 — CHANGES_REQUESTED** [review]

**Review by CJ42 — APPROVED** [review]

File: docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md [hunk 14]
```diff
@@ -356,9 +578,9 @@ _**MUST** be fired when the **[transfer](#transfer)** function gets executed suc
 
 | Name       | Type    | Description                                                                                                             |
 | :--------- | :------ | :---------------------------------------------------------------------------------------------------------------------- |
-| `operator` | address | The address of operator sending tokens.                                                                                 |
+| `operator` | address | The address of operator sender tokens.                                                                                 |
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md:581]: it seems strange, please revert the changes

File: docs/standards/smart-contracts/lsp7-digital-asset.md [hunk 15]
```diff
@@ -286,9 +502,9 @@ _**MUST** be fired when the **[transfer](#transfer)** function gets executed suc
 
 | Name       | Type    | Description                                                                                                                           |
 | :--------- | :------ | :------------------------------------------------------------------------------------------------------------------------------------ |
-| `operator` | address | The address of operator sending tokens.                                                                                               |
+| `operator` | address | The address of operator sender tokens.                                                                                               |
```
Comment (YamenMerhi) [inline docs/standards/smart-contracts/lsp7-digital-asset.md:505]: same here

**Review by YamenMerhi — APPROVED** [review]
